lib/stm32/common/timer_common_all.c

```diff
@@ -343,6 +343,21 @@ void timer_direction_down(uint32_t timer_peripheral)
 	TIM_CR1(timer_peripheral) |= TIM_CR1_DIR_DOWN;
 }
 
+/*---------------------------------------------------------------------------*/
+/** @brief Get the Timer count direction.
+
+Useful when the timer is in encoder interface mode and the DIR bit is modified
+by hardware.
+
+@param[in] timer_peripheral Unsigned int32. Timer register address base @ref
+tim_reg_base
+*/
+
+uint32_t timer_get_direction(uint32_t timer_peripheral)
```

Comment:
just make it return a bool. Document the return parameters, so people have a clue of knowing which is which.